### PR TITLE
Add Formula "nyancat"

### PR DESCRIPTION
**Pull request by yoshida-mediba**

### CODE REVIEW

File: Library/Formula/nyancat.rb [hunk 1]
```diff
@@ -0,0 +1,47 @@
+# Documentation: https://github.com/Homebrew/homebrew/blob/master/share/doc/homebrew/Formula-Cookbook.md
+#                /usr/local/Library/Contributions/example-formula.rb
+# PLEASE REMOVE ALL GENERATED COMMENTS BEFORE SUBMITTING YOUR PULL REQUEST!
```
Comment (bfontaine) [inline Library/Formula/nyancat.rb]: Please remove these comments.

File: Library/Formula/nyancat.rb [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+
+class Nyancat < Formula
+  desc "Nyancat Terminal"
+  homepage "https://github.com/klange/nyancat"
+  url "https://github.com/klange/nyancat/archive/0a7bfc16be777263841b512e1f2a9144afeb0752.tar.gz"
+  version "0a7bfc16be777263841b512e1f2a9144afeb0752"
```
Comment (bfontaine) [inline Library/Formula/nyancat.rb]: Please use tagged releases here.

File: Library/Formula/nyancat.rb [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+  sha256 "900da8bc30d443528534c31078dc25c9b5d73c851e761e288734ca5855f40870"
+
+  # depends_on "cmake" => :build
+  # depends_on :x11 # if your formula requires any X11/XQuartz components
```
Comment (bfontaine) [inline Library/Formula/nyancat.rb]: Please remove these comments.

File: Library/Formula/nyancat.rb [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+
+  def install
+    # ENV.deparallelize  # if your formula fails when building in parallel
+
+    # Remove unrecognized options if warned by configure
+    # system "./configure", "--disable-debug",
+    #                       "--disable-dependency-tracking",
+    #                       "--disable-silent-rules",
+    #                       "--prefix=#{prefix}"
+    # system "cmake", ".", *std_cmake_args
```
Comment (bfontaine) [inline Library/Formula/nyancat.rb]: Please remove these comments.

File: Library/Formula/nyancat.rb [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+    system "make" # if this fails, try separate make/make install steps
+
+    # Install Binary
```
Comment (bfontaine) [inline Library/Formula/nyancat.rb]: Please remove this comment.

File: Library/Formula/nyancat.rb [hunk 6]
```diff
@@ -0,0 +1,47 @@
...
+    bin.mkpath
```
Comment (bfontaine) [inline Library/Formula/nyancat.rb]: This line is not necessary.

File: Library/Formula/nyancat.rb [hunk 7]
```diff
@@ -0,0 +1,47 @@
...
+    bin.install "src/nyancat"
+
+    # Install Man
```
Comment (bfontaine) [inline Library/Formula/nyancat.rb]: Please remove this comment as well.

File: Library/Formula/nyancat.rb [hunk 8]
```diff
@@ -0,0 +1,47 @@
...
+    man.mkpath
```
Comment (bfontaine) [inline Library/Formula/nyancat.rb]: This line is not necessary.

File: Library/Formula/nyancat.rb [hunk 9]
```diff
@@ -0,0 +1,47 @@
...
+    # `test do` will create, run in and delete a temporary directory.
+    #
+    # This test will fail and we won't accept that! It's enough to just replace
+    # "false" with the main program this formula installs, but it'd be nice if you
+    # were more thorough. Run the test with `brew test nyancat`. Options passed
+    # to `brew install` such as `--HEAD` also need to be provided to `brew test`.
+    #
+    # The installed folder is not in the path, so use the entire path to any
+    # executables being tested: `system "#{bin}/program", "do", "something"`.
+    system "false"
```
Comment (bfontaine) [inline Library/Formula/nyancat.rb]: Please add a test.

File: Library/Formula/nyancat.rb [hunk 10]
```diff
@@ -0,0 +1,47 @@
...
+    # system "./configure", "--disable-debug",
+    #                       "--disable-dependency-tracking",
+    #                       "--disable-silent-rules",
+    #                       "--prefix=#{prefix}"
+    # system "cmake", ".", *std_cmake_args
+    system "make" # if this fails, try separate make/make install steps
+
+    # Install Binary
+    bin.mkpath
+    bin.install "src/nyancat"
```
Comment (bfontaine) [inline Library/Formula/nyancat.rb]: Is there a `make install` we can use?

File: Library/Formula/nyancat.rb [hunk 11]
```diff
@@ -0,0 +1,47 @@
...
+  def install
+    # ENV.deparallelize  # if your formula fails when building in parallel
+
+    # Remove unrecognized options if warned by configure
+    # system "./configure", "--disable-debug",
+    #                       "--disable-dependency-tracking",
+    #                       "--disable-silent-rules",
+    #                       "--prefix=#{prefix}"
+    # system "cmake", ".", *std_cmake_args
+    system "make" # if this fails, try separate make/make install steps
```
Comment (bfontaine) [inline Library/Formula/nyancat.rb]: Please remove the comment here.

**Comment by mikemcquaid** [comment]
The test is failing, would be good to work out where it's trying to write to that's invalid: http://bot.brew.sh/job/Homebrew%20Pull%20Requests/32193/version=yosemite/testReport/junit/brew-test-bot/yosemite/test_nyancat/

**Comment by yoshida-mediba** [comment]
It is similar to the sl command.
Fixed by reference to.

https://github.com/Homebrew/homebrew/blob/master/Library/Formula/sl.rb

File: Library/Formula/nyancat.rb [hunk 12]
```diff
@@ -0,0 +1,18 @@
...
+  homepage "https://github.com/klange/nyancat"
+  url "https://github.com/klange/nyancat/archive/0a7bfc16be777263841b512e1f2a9144afeb0752.tar.gz"
+  sha256 "900da8bc30d443528534c31078dc25c9b5d73c851e761e288734ca5855f40870"
+
+  head "https://github.com/klange/nyancat.git"
+
+  def install
+    system "make"
+    bin.install "src/nyancat"
+    man1.install "nyancat.1"
```
Comment (mikemcquaid) [inline Library/Formula/nyancat.rb]: Please use the `make install` and submit a patch to allow a custom prefix: https://github.com/klange/nyancat/blob/master/Makefile#L35-L37

File: Library/Formula/nyancat.rb [hunk 13]
```diff
@@ -0,0 +1,18 @@
...
+  end
+
+  test do
+    system "#{bin}/nyancat", "-h"
```
Comment (mikemcquaid) [inline Library/Formula/nyancat.rb]: Maybe `nyancat --frames 1` might provide a better test?

File: Library/Formula/nyancat.rb (hunk 12 above)
Comment (yoshida-mediba) [inline Library/Formula/nyancat.rb]: fix now
https://github.com/klange/nyancat/pull/34

**Comment by yoshida-mediba** [comment]
````
install src/nyancat /usr/local/Cellar/nyancat/0752/bin/nyancat
install: /usr/local/Cellar/nyancat/0752/bin/nyancat: No such file or directory
make: *** [install] Error 71
````

:disappointed:

File: Library/Formula/nyancat.rb [hunk 14]
```diff
@@ -0,0 +1,20 @@
...
+    sha256 "25cb3e15f05d8ada894df0653d37d22d927f1ddac1f2d6051ef79ea1d0ed4796"
+  end
+
+  def install
+    system "make"
+    system "make", "install", "instdir=#{prefix}"
+  end
+
+  test do
+    system "#{bin}/nyancat", "--frames 1"
```
Comment (bfontaine) [inline Library/Formula/nyancat.rb]: Use `"--frame", "1"`.

**Comment by bfontaine** [comment]
@yoshida-mediba I guess it should be `install -d` instead of `install`. `-d` creates the missing parent directories.

**Comment by yoshida-mediba** [comment]
Fixit!!

File: Library/Formula/nyancat.rb [hunk 15]
```diff
@@ -0,0 +1,21 @@
+class Nyancat < Formula
+  desc "Nyancat in your terminal, rendered through ANSI escape sequences."
+  homepage "https://github.com/klange/nyancat"
+  url "https://github.com/klange/nyancat/archive/0a7bfc16be777263841b512e1f2a9144afeb0752.tar.gz"
+  sha256 "900da8bc30d443528534c31078dc25c9b5d73c851e761e288734ca5855f40870"
+  version "0a7bfc1"
+
+  patch do
```
Comment (bfontaine) [inline Library/Formula/nyancat.rb]: Please add a comment explaining why this is necessary.

File: Library/Formula/nyancat.rb [hunk 16]
```diff
@@ -0,0 +1,21 @@
+class Nyancat < Formula
+  desc "Nyancat in your terminal, rendered through ANSI escape sequences."
+  homepage "https://github.com/klange/nyancat"
+  url "https://github.com/klange/nyancat/archive/0a7bfc16be777263841b512e1f2a9144afeb0752.tar.gz"
+  sha256 "900da8bc30d443528534c31078dc25c9b5d73c851e761e288734ca5855f40870"
+  version "0a7bfc1"
```
Comment (xu-cheng) [inline Library/Formula/nyancat.rb]: We need to use the release version.

File: Library/Formula/nyancat.rb (hunk 15 above)
Comment (mikemcquaid) [inline Library/Formula/nyancat.rb]: And use a PR url patch URL instead of a compare, thanks.

File: Library/Formula/nyancat.rb (hunk 16 above)
Comment (mikemcquaid) [inline Library/Formula/nyancat.rb]: e.g. a tagged release

**Comment by yoshida-mediba** [comment]
FIxit!!

File: Library/Formula/nyancat.rb [hunk 17]
```diff
@@ -0,0 +1,28 @@
+class Nyancat < Formula
+  desc "Nyancat in your terminal, rendered through ANSI escape sequences."
+  homepage "https://github.com/klange/nyancat"
+  url "https://github.com/klange/nyancat/archive/1.4.4.tar.gz"
+  version "1.4.4"
```
Comment (bfontaine) [inline Library/Formula/nyancat.rb]: You can remove this line; Homebrew gets it from the URL.

**Comment by yoshida-mediba** [comment]
Fix!!

**Comment by bfontaine** [comment]
Thank you @yoshida-mediba! :tada: 

For future reference, the preferred commit message format for new formulae is “`name version (new formula)`”, e.g. “`nyancat 1.4.4 (new formula)`”.

**Pull request closed without merging**